### PR TITLE
Database

### PR DESCRIPTION
**Pull request by ILabiak**
Wrote code to connect to database and to insert recipes, user data, categories and user favourite recipes. Also used parametrized queries to avoid sql injection

### CODE REVIEW

File: src/database/config/confix-example.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (akaeyuhi) [inline src/database/config/confix-example.json:1]: You can use dotenv module and .env file for storing configs

File: src/database/db.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
+'use strict';
+const mysql = require('mysql');
+const config = require('./config/config.json');
+
+let con;
```
Comment (akaeyuhi) [inline src/database/db.js:5]: You can return connection as a result of calling this function, but ok

File: src/database/db.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+(async () => {
+  await startDataBase();
+
+})();
+
+async function addUserToDb(username, email, password) {
+  const findUserRequest = `SELECT * FROM usersdata WHERE username = ?`;
+  const checkUserExistence = await sqlRequest(findUserRequest, [username]);
+  if (checkUserExistence[0]) return 'User with this username already exists';
```
Comment (akaeyuhi) [inline src/database/db.js:39]: throw error here

File: src/database/db.js [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+  const findMailRequest = `SELECT * FROM usersdata WHERE email = ?`;
+  const checkMailExistence = await sqlRequest(findMailRequest, [email]);
+  if (checkMailExistence[0]) return 'User with this email already exists';
```
Comment (akaeyuhi) [inline src/database/db.js:42]: and here

File: src/database/db.js [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+async function sqlRequest(sql, params = null) {
+  return new Promise((resolve, reject) => {
+    con.query(sql, params, (err, rows) => {
+      if (err) {
+        reject(new Error(err));
+      } else {
+        resolve(rows);
+      }
+    });
+  }).then();
```
Comment (akaeyuhi) [inline src/database/db.js:28]: You can replace then() with await i quess

Comment (akaeyuhi) [reply src/database/db.js:28]: return await new Promise...

**Review by akaeyuhi — CHANGES_REQUESTED** [review]
Some minor fixes needed

**Comment by ILabiak** [comment]
> 

Refactored code

**Review by akaeyuhi — APPROVED** [review]

**Pull request merged**